jsonrpc/src/main.ts

```diff
@@ -224,6 +255,7 @@ export namespace LogTraceNotification {
 
 export interface Tracer {
 	log(message: string, data?: string): void;
+	logLSP(message: string): void;
```

Comment:
This is a breaking API change. We should either make it optional or reuse the `log` signature. I looked at the client code and I was under the impression that the only difference is that the `log` prints [Trace - ...] and `logLSP` prints [LSP - ...]. Since the client knows the log format could this be decided on the client side.

Reply:
Actually we should have a log method that takes an JSON object literal. So the JSON RPC code shouldn't even convert it to a strings. Then someone could plug in a tracer that writes the JSON objects to a file.

Reply:
> Actually we should have a log method that takes an JSON object literal.

Wouldn't that be breaking as well?